### PR TITLE
feat : added cart UI and functionality

### PR DESCRIPTION
**Pull request by Priya-Patel31**
Files to check : 
* For context : /context/cart
* For UI : /pages/cart
* For API : /services/cart_services.js
* For generic function for calling api : /utils/fetchApi.js

You can avoid the CSS part as it is already reviewed .

### CODE REVIEW

File: src/shared/utils/fetchApi.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+
+async function fetchApi({ url, method, body, propertyName }) {
```
Comment (tushartiwari7) [inline src/shared/utils/fetchApi.js:3]: I liked the implementation of this util function.

File: src/context/cart/cartContext.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  const [{ cart, status }, dispatch] = useReducer(reducer, {
+    cart: [],
+    status: "IDLE",
+  });
+  const findItemInCart = (_id) => {
+    return cart.some((item) => {
+      return item._id === _id;
+    });
+  };
```
Comment (JantuDeb) [inline src/context/cart/cartContext.js]:
```
const findItemInCart = (_id) => cart.some((item) =>  item._id === _id);
```

File: src/shared/utils/fetchApi.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const defaultToken =
+    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiIxN2I2MDZjMS05OTk3LTRlNWYtODBmMC0xYWEyN2VlOGE1NTMiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20iLCJpYXQiOjE2NDc2MjgwODR9.CcWEd_xGSx_T1lCyeyyys4JLCVuWR5ThmVVtgi0qWmE";
+   const token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiIyYWNmMTkyZi0zYzk3LTQ1NWMtYjEzNi0yZmJkMzY0ODliYjEiLCJlbWFpbCI6InByaXlhQGdtYWlsLmNvbSIsImlhdCI6MTY0NzY5MzQxNX0.9vgp3G1Vrq-pHegSc4hfwA-i4EsKiKdpqPJT-ERY0IQ"
```
Comment (tushartiwari7) [inline src/shared/utils/fetchApi.js:6]: token will change on every user session so providing a defaultToken won't solve your problem. Try providing the token from localstorage only.

File: src/context/cart/cartContext.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        cartValue: curr.qty * curr.original_price + acc.cartValue,
+        totalDiscount:
+          curr.qty *
+            (curr.original_price -
+              price(curr.original_price, curr.discount_percent)) +
+          acc.totalDiscount,
+      };
+    },
+    { cartValue: 0, totalDiscount: 0 }
+  );
```
Comment (JantuDeb) [inline src/context/cart/cartContext.js:50]: This is really good use of !useReducer()`

**Review by tushartiwari7 — APPROVED** [review]
Really impressed by the number of good practices you used. 
Nothing major to fix just have a look at the suggestions and remove comments, empty files.

File: src/shared/utils/fetchApi.js (hunk 3 above)
Comment (Priya-Patel31) [reply src/shared/utils/fetchApi.js:6]: Yeah i will do  this when I will do the authentication .

File: src/pages/cart/components/cartItems/cartItems.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {HorizontalCard,Coupon,useCart} from "./index"
+
+const CartItems = () => {
+  const {cart} = useCart()
+  return (
+    <div className="cart-items-container">
+      <div className="cart-item">
+        {cart.map((item) => {
+         return <HorizontalCard key={item._id} {...item} />;
```
Comment (JantuDeb) [inline src/pages/cart/components/cartItems/cartItems.js]:
```
cart.map((item) =><HorizontalCard key={item._id} {...item} />)
```

File: src/pages/cart/cart.css [hunk 6]
```diff
@@ -0,0 +1,72 @@
+/* .carts-container{
+    background-color:var(--bg-color);
+    height:100vh;
+  } */
```
Comment (t007rushi) [inline src/pages/cart/cart.css:4]: remove commented code

**Review by JantuDeb — APPROVED** [review]

File: src/pages/productListing/components/filter/filter.js [hunk 7]
```diff
@@ -27,81 +41,127 @@ const Filter = () => {
...
             <input
               type="radio"
-              value="price-low-to-high"
+              value="LOW_TO_HIGH"
+              checked={state.sortBy === "LOW_TO_HIGH"}
               name="price"
               id="price-low-to-high"
+              onChange={(e) =>
+                dispatch({ type: "SORT", payload: { value: e.target.value } })
+              }
```
Comment (t007rushi) [inline src/pages/productListing/components/filter/filter.js:50]: you can put in single component to follow DRY

File: src/services/cart_services.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import fetchApi from "../shared/utils/fetchApi";
```
Comment (t007rushi) [inline src/services/cart_services.js:1]: nicely done seperate out the api calls

**Review by t007rushi — APPROVED** [review]
some changes else clean code writing

File: src/context/cart/cartContext.js (hunk 4 above)
Comment (Priya-Patel31) [reply src/context/cart/cartContext.js:50]: thank you !!

File: src/shared/utils/fetchApi.js (hunk 1 above)
Comment (Priya-Patel31) [reply src/shared/utils/fetchApi.js:3]: thank you

File: src/pages/cart/cart.css (hunk 6 above)
Comment (Priya-Patel31) [reply src/pages/cart/cart.css:4]: yes I will remove it.

File: src/pages/cart/components/cartItems/cartItems.js (hunk 5 above)
Comment (Priya-Patel31) [reply src/pages/cart/components/cartItems/cartItems.js]: done

File: src/context/cart/cartContext.js (hunk 2 above)
Comment (Priya-Patel31) [reply src/context/cart/cartContext.js]: done

File: src/pages/productListing/components/filter/filter.js (hunk 7 above)
Comment (Priya-Patel31) [reply src/pages/productListing/components/filter/filter.js:50]: yes I will try to do .

**Pull request merged**